### PR TITLE
Automatically formatting and cleaning code on commit

### LINKED ISSUE #559: Add automated prettier command
### Problem

Files continue to have inconsistent formatting since not all PRs apply the prettier logic.

### Solution

Automate the prettier command run.
Consider something like [husky precommit](https://prettier.io/docs/en/precommit.html)

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:
This PR adds automatically formatting and cleaning code on commit.
Code style is standard for typescript. All files changed prior commit will be formatted.

**Related issue(s)**:

Fixes #559 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,5 +75,21 @@
         "protobufjs": "^7.2.4",
         "semver": "^7.5.3",
         "nx": "16.5.5"
-    }
+    },
+    "lint-staged": {
+        "packages/**/src/**/*.ts": [
+          "prettier --write"
+        ],
+        "!(packages/**/src/dist/)*.ts": [
```
Comment (Nana-EC) [inline package.json:83]: We don't check dist in, is this just a nice to have for local troubleshooting view?

File: package.json [hunk 2]
```diff
@@ -71,5 +75,21 @@
...
+            "prettier --write"
+        ],
+        "!(packages/**/src/node_modules/)*.ts": [
```
Comment (Nana-EC) [inline package.json:86]: Same as above, is this just a nice to have for local troubleshooting view?

File: package.json (hunk 1 above)
Comment (Nana-EC) [inline package.json:83]: No such dir. Did you mean?
```suggestion
        "!(packages/**/dist/)*.ts": [
```

**Review by Nana-EC — COMMENTED** [review]
Nice. Some questions

File: package.json (hunk 1 above)
Comment (georgi-l95) [reply package.json:83]: Yes, I think we don't need to check and format anything under dist folder. Also do you think we should also include tests to be properly formated upon changes and commits ? 
```
        "packages/**/tests/**/*.ts": [
          "prettier --write"
        ],
```

File: package.json (hunk 2 above)
Comment (georgi-l95) [reply package.json:86]: Yes, i believe so.

File: package.json (hunk 1 above)
Comment (georgi-l95) [reply package.json:83]: Ah, yes you're right. I've typed the wrong dir.

Comment (Nana-EC) [reply package.json]: Yeah we should format test files also

Comment (georgi-l95) [reply package.json]: Added them as well.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**